### ISSUE TITLE
Inconsistent and confusing CryptoEngine related things

### ISSUE THREAD

**Issue by ksachdeva** [PeculiarVentures/PKI.js#132]
Hi @YuryStrozhevsky,

I was trying to write a simple nodejs application (based on NodePKCS12Example) and found certain things bit confusing.

Here is the sample application https://github.com/ksachdeva/pkijs-node-example . The sample works and instructions on how to run it are in the ReadMe.

The sample is written using typescript and is essentially a port of your CertificateCompleteExample. In brief, I am creating a certificate and private key. 

From what I understood after reading your code is that in order to run this sample using nodejs I needed to provided two things:

- A crypto implementation
- An implementation of WebCrypto.Subtle interface

I borrowed the crypto implementation from NodePKCS12Example and for WebCrypto.Subtle I am using https://github.com/PeculiarVentures/node-webcrypto-ossl

The setEngine method is used to provide the CryptoEngine implementation. 

Here are my questions -

- I had to figure out how to properly pass the parameters to setEngine method after reading the implementation of various methods in common.js file. Here is the snippet from my example:

```
const webcrypto = new WebCrypto();

setEngine('nodeEngine', nodeSpecificCrypto, new CryptoEngine({
    crypto: nodeSpecificCrypto,
    subtle: webcrypto.subtle,
    name: 'nodeEngine'
}));
```

Now signature of setEngine method is `setEngine(name, crypto, subtle)` where as in order to make it work I had to pass CryptoEngine object as the third parameter (i.e. subtle). I figured this by seeing the implementation of various methods in common.js

It is bit strange but may be I am missing something. Please educate.

- In my example, I managed to use the CryptoEngine implementation from your library and did not have to write NodeCryptoEngine as you have done in NodePKCS12Example. So far based on what I see only thing required is NodeSpecificCrypto implementation. Please suggest when would one need the NodeCryptoEngine ?

Sorry for a long description. Thanks again for the stellar work on this library and hope to get guidance on the questions asked.

Regards
Kapil

**Comment by YuryStrozhevsky** [comment]
@ksachdeva The "strange things" were implemented in scope of PKIjs v2.1 version. Starting this version PKIjs would be not a "WebCrypto API lirary", but an universal library which could be run in almost any environment. We will not consider PKIjs running only in browser, we will not consider PKIjs running only in "WebCrypto-friendly" environment - no, now PKIjs could be running on top of any crypto-based API.

**Comment by ksachdeva** [comment]
@YuryStrozhevsky Thanks.
